### PR TITLE
refactor: implemented findPath function to replace require.resolve

### PR DESCRIPTION
**Pull request by lee-tammy**

### CODE REVIEW

File: src/package-tree.ts [hunk 1]
```diff
@@ -166,35 +161,50 @@ export async function resolvePaths(
...
+export async function findPath(
+    mod: string, parentPath: string): Promise<string> {
+  const moduleFolder = path.join(parentPath, 'node_modules');
+  try {
+    const filesInFolder = await util.readdir(moduleFolder);
+    if (filesInFolder.includes(mod)) {
+      return path.join(moduleFolder, mod);
     }
+  } catch (err) {
   }
```
Comment (lee-tammy) [inline src/package-tree.ts]: Is there a way to check if a file/directory exists without having to use try/catch?

Comment (kjin) [reply src/package-tree.ts]: I would suggest encapsulating the "try stat or readdir, and catch it" into its own utility function, returning a boolean value based on whether an error was thrown or not.

File: src/package-tree.ts [hunk 2]
```diff
@@ -125,7 +120,7 @@ export async function getPackagePOIList(path: string):
   const files = await getJSFiles(path);
 
   await Promise.all(files.map(async (file) => {
-    const content = await readFile(file, 'utf8');
+    const content = await util.readFile(file, 'utf8');
```
Comment (kjin) [inline src/package-tree.ts:122]: Should we be using `customReadFilep` here?

Comment (kjin) [reply src/package-tree.ts:122]: In addition, there may need to exist some type of mock for `readdir` as well. Though, things will get unwieldy if we are adding too many `custom*` arguments.

File: src/util.ts [hunk 3]
```diff
@@ -17,6 +17,15 @@
...
 import pify from 'pify';
 
-export const filesInDir = pify(fs.readdir);
+export const readdir = pify(fs.readdir);
 export const readFile = pify(fs.readFile);
-export const fileInfo = pify(fs.stat);
+export const stat = pify(fs.stat);
+
+export async function exists(file: string){
+    try{
```
Comment (kjin) [inline src/util.ts]: Looks like you need to run `npm run fix`.

File: src/package-tree.ts (hunk 1 above)
Comment (lee-tammy) [reply src/package-tree.ts]: done

File: src/package-tree.ts [hunk 4]
```diff
@@ -51,8 +46,8 @@ export interface PackageTree<T = null> {
  * Takes in the root directory of a project and returns returns a PackageTree
  */
 export async function generatePackageTree(
-    rootDir: string, customReadFilep?: ReadFileP): Promise<PackageTree> {
-  customReadFilep = customReadFilep || readFilep;
+    rootDir: string, customReadFilep?: ReadFileP): Promise<PackageTree<null>> {
+  customReadFilep = customReadFilep || util.readFile;
```
Comment (jplaisted) [inline src/package-tree.ts]: There is an easier way of defaulting parameters. You can just set the variable in the parameter to the value.

`function example(a: number = 0)` will default `a` to `0` if not passed.

File: src/util.ts [hunk 5]
```diff
@@ -17,6 +17,15 @@
...
+        await stat(file);
+    }catch(err){
+        return false;
```
Comment (jplaisted) [inline src/util.ts]: Please check the specific error code is `ENOENT`. If it isn't then rethrow the error.

https://nodejs.org/api/errors.html#errors_common_system_errors

File: src/package-tree.ts (hunk 2 above)
Comment (lee-tammy) [reply src/package-tree.ts:122]: Decided to stop using custom arguments to test

File: src/package-tree.ts (hunk 4 above)
Comment (lee-tammy) [reply src/package-tree.ts]: cool, I changed it!

File: src/util.ts (hunk 5 above)
Comment (lee-tammy) [reply src/util.ts]: done

File: src/util.ts [hunk 6]
```diff
@@ -16,16 +16,21 @@
...
-export async function exists(file: string){
-    try{
-        await stat(file);
-    }catch(err){
-        return false;
+export async function exists(file: string) {
+  try {
+    await stat(file);
+  } catch (err) {
+    if(err === ENOENT){
```
Comment (jplaisted) [inline src/util.ts]: Nit: run the formatter

Comment (lee-tammy) [reply src/util.ts]: done

**Review by kjin — APPROVED** [review]
LGTM

**Pull request merged**